app/src/main/java/com/android/streetworkapp/ui/progress/Progression.kt

```diff
@@ -75,6 +75,7 @@ fun ProgressScreen(
   val currentProgression by progressionViewModel.currentProgression.collectAsState()
 
   currentUser?.uid?.let { progressionViewModel.getOrAddProgression(it) }
+  currentUser?.let { progressionViewModel.checkScore(it.score) }
```

Comment:
There's an issue here. Imagine a user collects enough point to pass 2 levels, this would only grant him the first achievement and not the second one. If we still want to keep the user viewmodel and progression viewmodel we should have a loop in the checkscore that loops until the user has been granted all achievements

Reply:
Good catch, I added the loop in `checkScore` (in the ViewModel, since it makes more sens than in the UI. Thank you!

Reply:
There's a slight modification I would make to the loop: instead of hardcoding platinum as a condition, you could add a ACHIEVEMENT_CEILING (or something like that) to the enum. Then you could just fetch the max achievement by taking ordinal ACHIEVEMENT_CEILING - 1. This way if we add ranks, everything still works

Reply:
I am not 100% sure that I see how you imagine it, but I updated the progression with a CEILING value. You can check it and tell me if you disagree.